assignments/array-methods.js

```diff
@@ -2,82 +2,508 @@
+    email: "amacgiollapheadair1b@xing.com",
+    shirt_size: "2XL",
+    company_name: "Kwimbee",
+    donation: 214
+  },
+  {
+    id: 49,
+    first_name: "Bel",
+    last_name: "Alway",
+    email: "balway1c@ow.ly",
+    shirt_size: "S",
+    company_name: "Voolia",
+    donation: 107
+  },
+  {
+    id: 50,
+    first_name: "Shell",
+    last_name: "Baine",
+    email: "sbaine1d@intel.com",
+    shirt_size: "M",
+    company_name: "Gabtype",
+    donation: 171
+  }
+];
 
 // ==== Challenge 1: Use .forEach() ====
-// The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
+// The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName.
 let fullName = [];
+runners.forEach(function(runner) {
+  fullName.push(`${runner.first_name} ${runner.last_name}`);
+});
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
 let allCaps = [];
-console.log(allCaps); 
+const firstName = runners.map(function(runner) {
+  return allCaps.push(toUpper(runner.first_name));
+});
 
+function toUpper(runner) {
+  return runner.toUpperCase();
+}
+
+console.log(allCaps);
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
 let largeShirts = [];
-console.log(largeShirts);
 
+function runnersWithLargeShirts() {
+  const largeShirt = runners.filter(runner => runner.shirt_size === "L");
+
+  largeShirts.push(largeShirt);
+  return largeShirts;
+}
+
+console.log(runnersWithLargeShirts());
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
 let ticketPriceTotal = [];
+const sumOfTickets = runners.reduce(
+  (total, runners) => total + runners.donation,
+  0
+);
+ticketPriceTotal.push(sumOfTickets);
 console.log(ticketPriceTotal);
-
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
 
-// Problem 1
+// Problem 1. The community gives a special mention to the donors of more than 250.
+let greatDonors = [];
+greatDonors = runners.filter(runner => runner.donation > 250);
```

Comment:
same as above you don't need to initialise this as an array `let greatDonors = [];` you can just just do `let greatDonors = runners.filter(runner => runner.donation > 250);`